src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name);
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
```

Comment:
Same here

Reply:
Sure.
I understand.
I have removed the constructor from Carnivore, and he still inherits the constructor from its parent class.